### ISSUE TITLE
Problems/Suggestions

### ISSUE THREAD

**Issue by MFlisar** [deano2390/MaterialShowcaseView#8]
Instead of writting more posts I summurize my ideas, feel free to ignore them...

**PROBLEMS**
* `MaterialShowcaseSequence` is prone to errors, as it can happen that the user touches the layout underneath while going from one `Showcase` to another... Can be solved if I add a transparent view underneath that consumes all touches and is removed after the sequence is finished
* before creating a new bitmap, check if the bitmap != null and recycle it... I already had a memory problem and I suppose that's the cause. Although I've not investigated that to much yet... I don't know if  `canvas.drawRect` is better than creating a `bitmap` for the color background, but probably it is. But I'm not sure. So you could get rid of the bitmap...

**SUGGESTION**
* I personally would like to have more possibilities to highlight the targets. I would suggest a circle and a rectangle (with adjustable borders?)
* Maybe some navigation like next/previous? And the possibility to add a "Don't show again" button at the end of a sequence?
* in a sequence a transformation animation between `Showcases` would look nice in my opinion...

**Comment by deano2390** [comment]
Very good points. The point about the user touching between showcases is a big issue which I intend to tackle soon. The transition between them could well be the answer.

**Comment by MFlisar** [comment]
Yeah, people like to "jump" over the tutorials, so I think in practice, this problem will occur very often...

**Comment by MFlisar** [comment]
I've some experiences with it now and I'm very sure, that the bitmap creates memory issues if using the showcase heavily and over and over again... Recycling it probably solves it. I've not checked the complete library, if you use bitmaps somewhere else as well, this clean up should be done there as well...

Maybe, as I said, you could even get rid of the bitmap at all...

**Comment by deano2390** [comment]
Good spot sir, the memory usage was indeed creeping up with every new showcase view. I've fixed it v1.0.3

**Comment by MFlisar** [comment]
Thanks. You introduced a small bug though... `MaterialShowcaseView.notifyOnDismissed()` will throw a `NullPointerException` if `mListeners==null which will happen after the new clean up as you set it null after clearing it...

**Comment by deano2390** [comment]
Good shout, fixed it. You are like my personal QA team.

**Comment by MFlisar** [comment]
Seems so ;-). I would check out the project and fix the small things myself, but I currently can't. I could only do so on the weekend...

**Comment by Zeliret** [comment]
I would glad to see a rectangle too! :P
